clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.groovy

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.jobs.local
+
+import com.netflix.spinnaker.clouddriver.jobs.JobExecutor
+import com.netflix.spinnaker.clouddriver.jobs.JobRequest
+import com.netflix.spinnaker.clouddriver.jobs.JobStatus
+import groovy.util.logging.Slf4j
+import org.apache.commons.exec.*
+import org.springframework.beans.factory.annotation.Value
+import rx.Scheduler
+import rx.functions.Action0
+import rx.schedulers.Schedulers
+
+import java.util.concurrent.ConcurrentHashMap
+
+@Slf4j
+class JobExecutorLocal implements JobExecutor {
+
+  @Value('${jobs.local.timeoutMinutes:10}')
+  long timeoutMinutes
+
+  Scheduler scheduler = Schedulers.computation()
+  Map<String, Map> jobIdToHandlerMap = new ConcurrentHashMap<String, Map>()
+
+  @Override
+  String startJob(JobRequest jobRequest) {
+    log.info("Starting job: $jobRequest.tokenizedCommand...")
+
+    String jobId = UUID.randomUUID().toString()
+
+    scheduler.createWorker().schedule(
+      new Action0() {
+        @Override
+        public void call() {
+          ByteArrayOutputStream stdOutAndErr = new ByteArrayOutputStream()
+          PumpStreamHandler pumpStreamHandler = new PumpStreamHandler(stdOutAndErr)
+          CommandLine commandLine
+
+          if (jobRequest.tokenizedCommand) {
+            log.info("Executing $jobId with tokenized command: $jobRequest.tokenizedCommand")
+
+            // Grab the first element as the command.
+            commandLine = new CommandLine(jobRequest.tokenizedCommand[0])
+
+            // Treat the rest as arguments.
+            String[] arguments = Arrays.copyOfRange(jobRequest.tokenizedCommand.toArray(), 1, jobRequest.tokenizedCommand.size())
+
+            commandLine.addArguments(arguments, false)
+          } else {
+            log.info("No tokenizedCommand specified for $jobId.")
```

Comment:
You don't need to log this here, since it the exception below will make it into the logs.